aries_cloudagent/transport/inbound/queue/loader.py

```diff
@@ -0,0 +1,26 @@
+"""Dynamic loading of pluggable inbound queue engine classes."""
+import logging
+from typing import Optional, cast
+
+from ....core.profile import Profile
+from ....utils.classloader import ClassLoader
+from .base import BaseInboundQueue, InboundQueueConfigurationError
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_inbound_queue(root_profile: Profile) -> Optional[BaseInboundQueue]:
+    """Given settings, return instantiated inbound queue class."""
+    class_path = root_profile.settings.get("transport.inbound_queue_class")
+    if not class_path:
+        LOGGER.info("No inbound queue loaded")
+        return None
+    class_path = cast(str, class_path)
+    klass = ClassLoader.load_class(class_path)
+    instance = klass(root_profile)
+    if not isinstance(instance, BaseInboundQueue):
+        raise InboundQueueConfigurationError(
+            "Configured class is not a subclass of BaseInboundQueue"
+        )
+    return instance
```

Comment:
The `get_outbound_queue` (and now `get_inbound_queue`) method has always felt a bit out of place to me. I wonder if it would make sense to instead model this as a new type of `Provider` that reads from settings to determine class path. Getting an instance would then be done by injection (using `inject_or` to permit the return of a `None`). This would hopefully provide a useful interface for future use while reducing duplicated code.

This is just a suggestion, though. I don't consider this to be a blocker for this PR.